### PR TITLE
Order.AddOrderItem Clean Code Refactor

### PR DESCRIPTION
**Pull request by Bruellhusten**
Clean Code Refactor
Naming => _AddOrderItem_ also updated the OrderItem => misinformation
Added _OrderItemDTO_ to replace excessive method parameters 
_PutOrderItem_ may not be the ideal method naming. Alternative would be _AddOrUpdateOrderItem_.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet-architecture/eShopOnContainers?pullRequest=1523) <br/>All CLA requirements met.

File: src/Services/Ordering/Ordering.Domain/AggregatesModel/OrderAggregate/Order.cs [hunk 1]
```diff
@@ -72,29 +73,38 @@ protected Order()
         // This Order AggregateRoot's method "AddOrderitem()" should be the only way to add Items to the Order,
         // so any behavior (discounts, etc.) and validations are controlled by the AggregateRoot 
         // in order to maintain consistency between the whole Aggregate. 
-        public void AddOrderItem(int productId, string productName, decimal unitPrice, decimal discount, string pictureUrl, int units = 1)
+        public void PutOrderItem(OrderItemDto orderItemToAdd)
```
Comment (sughosneo) [inline src/Services/Ordering/Ordering.Domain/AggregatesModel/OrderAggregate/Order.cs:76]: Wouldn't be better to have method name as `AddOrUpdateOrderItem` ?

File: src/Services/Ordering/Ordering.Domain/AggregatesModel/DTOs/OrderItemDto.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Ordering.Domain.AggregatesModel.DTOs
+{
+    public class OrderItemDto
```
Comment (sughosneo) [inline src/Services/Ordering/Ordering.Domain/AggregatesModel/DTOs/OrderItemDto.cs:7]: Wouldn't be better to keep `OrderItemDto` file outside of `AggregatesModel` but within the `Ordering.Domain` project ?
That way the DTO stays close to the Order aggregate model but stays separate than AggregatesModel. Thoughts ?

**Comment by sughosneo** [comment]
> Clean Code Refactor
> Naming => _AddOrderItem_ also updated the OrderItem => misinformation
> Added _OrderItemDTO_ to replace excessive method parameters
> _PutOrderItem_ may not be the ideal method naming. Alternative would be _AddOrUpdateOrderItem_.

Hi @Bruellhusten , thank you for submitting this PR. 
Yes, I would also prefer to go with `AddOrUpdateOrderItem` name instead.

**Comment by sughosneo** [comment]
Hi @Bruellhusten, `dev` branch of eShop reference application has now been updated to `.NET 5`. 
If you could take the latest and update this PR. So that we can review it further. 

Thank you.

**Comment by sughosneo** [comment]
As this PR is stale for sometime, I am closing it now. Please take the latest from the `dev` branch and submit a fresh PR with suggested changes. We will review and merge the changes. Thank you.

**Pull request closed without merging**